pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -18,23 +18,36 @@ func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.
 		}
 		return oidc.ClientContext(ctx, certKeyClient), nil
 	}
+
 	return ctx, nil
 }
 
 func GetClientWithCertKey(httpClient *http.Client, certificate, key string) error {
-	if certificate != "" && key != "" {
-		keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
-		if err != nil {
-			return err
-		}
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM([]byte(certificate))
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs:      caCertPool,
-				Certificates: []tls.Certificate{keyPair},
-			},
-		}
+	tlsConfig, err := getTLSClientConfig(certificate, key)
+	if err != nil {
+		return err
 	}
+
+	// Use the default transport to pick up on things like http.ProxyFromEnvironment
+	transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
I think this is fine but let me get back to you real quick whether we're okay with changing that behavior. This new behavior might break installations if they had a proxy configured but relied on proxy settings not being followed for keycloak.

Reply:
My main concern with not having the proxy settings is that this capability is now non-functional in scenarios where users can _only_ resolve provider endpoints through a proxy. It's also misleading because Rancher provides specific helm values for `proxy` and `noProxy` so users could be left debugging network-related issues when all the while the issue is in the underlying code that just isn't listening to the settings they're giving it.

If this PR needs to wait for a more appropriate version bump or the proxy corrections need to wait, I get it, but I do think this is an important setting to get through because I know (within weeks) I personally will be in that exact scenario where my auth provider can only be reached through the proxy I give rancher.

Reply:
We definitely want this to work with the proxy settings, but _as is_ this is a breaking change. I think we could make it work without the breaking change, but I'd prefer we address that in a separate issue/PR to figure out the branch/timing here. Do you mind creating a separate issue/PR for this? This way the current PR can stay focused on the CA change.